### DISCUSSION TITLE
eval 컨텍스트에 대하여

### DISCUSSION THREAD

**Discussion by Ham-mi** [read-modern-js/deep-dive#34] [General]
21장에서 eval 함수 사용을 금지해야한다고 하는데 23장 실행 컨텍스트에서 eval 실행 컨텍스트가 따로 있는 이유가 왜 일까요? 사용에 이유가 있을까요?

**Comment by yanggak12** [comment]
소스코드를 4가지 타입으로 구분하는 이유는 소스코드의 타입에 따라 실행 컨텍스트를 생성하는 과정과 관리 내용이 다르기 때문입니다.
eval코드는 strict mode에서 자신만의 독자적인 스코프를 생성해서 eval코드가 평가되면 eval실행 컨텍스트가 생성이 됩니다!
즉, eval 코드는 eval 실행 컨텍스트를 생성합니다.

**Comment by gangmin2** [comment]
eval 함수 사용을 금지해야 한다고는 하지만 아직도 사용되는 부분이 있을 거고 그래서 아직도 eval 실행 컨텍스트가 따로 존재하고 있는 게 아닌가 생각됩니다!
